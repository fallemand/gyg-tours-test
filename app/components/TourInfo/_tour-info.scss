@import '../../styles/variables';
@import './tour-info__small';

@media ( min-width: $breakpoint-desktop ) {
  @import './tour-info__medium';
}
