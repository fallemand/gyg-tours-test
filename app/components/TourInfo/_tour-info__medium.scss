.tour-info {
  text-align: left;
  padding: 15px 20px;
  margin: 15px 0;
  cursor: pointer;
  transition: box-shadow 0.4s ease;
  box-shadow: 0 0 2px rgba(0,0,0,0.2);
  position: relative;

  &:hover {
    box-shadow: 0 0 8px rgba(0,0,0,0.35);
  }

  .tour-info__logo {
    display: inline-block;
    max-width: 210px;
    width: 20%;
  }

  .tour-info__content {
    display: inline-block;
    width: 70%;
    padding: 0 20px;
    vertical-align: top;
  }

  .tour-info__title {
    margin: 0 20px 0 0;
    display: inline-block;
  }

  .tour-info__rating {
    margin-right: 0;
  }

  .tour-info__special-offer {
    position: absolute;
    right: -5px; top: -5px;
    z-index: 1;
    overflow: hidden;
    width: 75px; height: 75px;
    text-align: right;
    margin: 0;
    padding: 0;
    background-color: transparent;

    .tour-info__special-offer-message {
      font-size: 9px;
      font-weight: bold;
      color: #FFF;
      text-transform: uppercase;
      text-align: center;
      line-height: 20px;
      transform: rotate(45deg);
      -webkit-transform: rotate(45deg);
      width: 100px;
      display: block;
      background: #79A70A;
      background: linear-gradient(#FF5534 0%, #D9482C 100%);
      box-shadow: 0 3px 10px -5px rgba(0, 0, 0, 1);
      position: absolute;
      top: 19px; right: -21px;

      &:before {
        content: "";
        position: absolute; left: 0px; top: 100%;
        z-index: -1;
        border-left: 3px solid #D9482C;
        border-right: 3px solid transparent;
        border-bottom: 3px solid transparent;
        border-top: 3px solid #D9482C;
      }

      &:after {
        content: "";
        position: absolute; right: 0px; top: 100%;
        z-index: -1;
        border-left: 3px solid transparent;
        border-right: 3px solid #D9482C;
        border-bottom: 3px solid transparent;
        border-top: 3px solid #D9482C;
      }
    }
  }
}
