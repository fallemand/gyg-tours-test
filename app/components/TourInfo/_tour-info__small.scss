.tour-info {
  text-align: center;
  border-radius: 2px;
  padding: 15px 10px;
  margin: 15px 0;

  .tour-info__logo {
    max-width: 100px;
  }

  .tour-info__title {
    font-size: $font-size-18;
    color: $font-color-primary;
    margin: 5px;
  }

  .tour-info__rating {
    font-size: 14px;
    color: $font-color-secondary;
    margin-right: 10px;

    small {
      color: $font-color-secondary;
      font-size: $font-size-12;
    }
  }
}
