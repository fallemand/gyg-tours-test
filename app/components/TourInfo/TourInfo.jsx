import React from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';
import Rating from '../Rating';
import './_tour-info.scss';

const TourInfo = ({
  className, image, title, rating, price, currency, isSpecialOffer, onClick,
}) => (
  <div onClick={onClick} className={classnames(className, 'tour-info')}>
    <img className="tour-info__logo" src={image} alt="Tour Logo" />
    <div className="tour-info__content">
      <h2 className="tour-info__title">{title}</h2>
      <Rating className="tour-info__rating" value={parseFloat(rating)} />
      <p className="tour-info__price">
        <small className="tour-info__price-label">From</small>
        {currency}
        {price}
      </p>
      { isSpecialOffer && (
        <div className="tour-info__special-offer">
          <span className="tour-info__special-offer-message">Special Offer!</span>
        </div>)
      }
    </div>
  </div>
);

TourInfo.propTypes = {
  className: PropTypes.string,
  onClick: PropTypes.func,
  image: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  price: PropTypes.string.isRequired,
  currency: PropTypes.string.isRequired,
  isSpecialOffer: PropTypes.bool.isRequired,
  rating: PropTypes.string.isRequired,
};

TourInfo.defaultProps = {
  className: null,
  onClick: null,
};

export default TourInfo;
