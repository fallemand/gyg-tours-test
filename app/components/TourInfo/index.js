import TourInfo from './TourInfo';

export default TourInfo;
