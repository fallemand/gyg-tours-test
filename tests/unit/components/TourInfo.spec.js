import React from 'react';
import TourInfo from '../../../app/components/TourInfo';

describe('RestaurantInfo', () => {
  const props = {
    image: '__LOGOURI__',
    title: '__TITLE__',
    rating: '__RATING__',
    location: '__LOCATION__',
    price: '__PRICE__',
    currency: '__CURRENCY__',
    isSpecialOffer: true,
    className: '__CLASSNAME__',
  };

  it('render component', () => {
    const component = shallow(<TourInfo {...props} />);
    expect(escapeSnapshot(component)).toMatchSnapshot();
  });
});
