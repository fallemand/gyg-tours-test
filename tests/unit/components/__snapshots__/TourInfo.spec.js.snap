// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestaurantInfo render component 1`] = `
"<div onClick={{...}} className='__CLASSNAME__ tour-info'>
  <img className='tour-info__logo' src='__LOGOURI__' alt='Tour Logo' />
  <div className='tour-info__content'>
    <h2 className='tour-info__title'>
      __TITLE__
    </h2>
    <Rating className='tour-info__rating' value={NaN} />
    <p className='tour-info__price'>
      <small className='tour-info__price-label'>
        From
      </small>
      __CURRENCY__
      __PRICE__
    </p>
    <div className='tour-info__special-offer'>
      <span className='tour-info__special-offer-message'>
        Special Offer!
      </span>
    </div>
  </div>
</div>"
`;
